src/backend/services/ExtractZipService.ts

```diff
@@ -22,6 +30,11 @@ enum ExtractionValidation {
   VALID = 'VALID'
 }
 
+type extractOptions = {
+  deleteOnEnd?: boolean
+  preserveStructure?: boolean
```

Comment:
nitpick. I think `preserveStructure` is the wrong name for that functionality. all it is doing is copying a subfolder with the same name as the zip up a level. this isn't general functionality and is only needed for the chromie craft zip. I think we may just want to make it a separate function independent from the extract zip service

Reply:
Done. Moved the logic to the prepareBaseGameForModing  method